event_attribution_reporting.md

```diff
@@ -0,0 +1,136 @@
+
+The additional restrictions in this proposal represent one potential privacy vs. utility tradeoff, and may be adjusted to provide more utility or privacy.
+
+This proposal should be read as an optional additive component to the Click Through Attribution Reporting API.
+
+
+## API Changes
+
+### Registering attribution sources without a click
+
+Anchor elements will expose a new boolean `registerattributionsource` attribute which when added to the tag, registers an attribution source event similar to when an attribution source anchor tag is clicked.
+
+
+### Registering attribution sources with JavaScript
+
+Scripts running on a source site can register attribution sources via a new JavaScript API without needing to add anchor elements to the page.
+
+The browser will expose a new interface:
+
+```
+[Exposed=Window] interface Attribution {
+  bool registerAttributionSource(AttributionSourceParams params);
+}
+```
+
+See [this section](https://github.com/WICG/conversion-measurement-api/blob/main/README.md#registering-attribution-sources-for-windowopen-navigations) of the Attribution Reporting explainer for the definition of `AttributionSourceParams`. When invoked the browser will directly add the specified source to storage, and return whether the browser was successful in parsing the params.
+
+
+### Different Classes of Attribution Sources
+
+Attribution sources will have a browser associated `source_type`. Attribution sources registered with the `registerattributionsource` attribute or via the JS API  will have source_type set to "event". Sources registered via clicks on anchor tags or with the `window.open()` API will have `source_type` set to "navigation".
+
+
+### Attribution Triggering
+
+Attribution trigger redirects will be able to specify different data to be used, depending on the `source_type` of any associated sources.
+
+This is done via a new query param, `event-source-trigger-data` on the trigger [redirect](https://github.com/WICG/conversion-measurement-api#triggering-attribution). This data will be limited to 1 bit. This data will be noised 5% of the time, matching the [noise](https://github.com/WICG/conversion-measurement-api#data-limits-and-noise) for navigation sources.
+
+Event sources will only be able to generate a single attribution report, compared to the three reports for navigation sources. Alongside the 1 bit of data, this further restricts the total amount of trigger side data which can be associated with a single `attributionsourceeventid`. See [privacy considerations](#privacy-considerations) for why this is necessary.
+
+
+### Coarse Attribution Expiry
+
+Event sources will have their `attributionexpiry` rounded to the nearest whole day.
+
+The `attributionexpiry` of a source can be used to associate attributionsourceeventids with a coarse trigger side timestamp. Because event sources are easier to register en masse than navigation sources, it would allow a reporting origin to do this with a finer granularity.
```

Comment:
```suggestion
The `attributionexpiry` of a source can be used to associate `attributionsourceeventid`s with a coarse trigger side timestamp. Because event sources are easier to register en masse than navigation sources, it would allow a reporting origin to do this with a finer granularity.
```